src/pipe.js

```diff
@@ -4,6 +4,7 @@
     var scripts = doc.getElementsByTagName('script');
     // State to maintain if all fragments on the page are initialized
     var initState = [];
+    var ranges = [];
```

Comment:
We can do this by introducing a unique identifier to fragment class. Previously id was unique to a fragment, Now its modified based on the index.